org/all-prs.js

```diff
@@ -9,106 +9,126 @@
 
 //eslint-disable-next-line no-console
 const ok = test => console.info(`${test} OK`);
 
 // # Commits
 // Rules related to commit descriptions and body
 
-// TODO: consider using https://github.com/marionebl/commitlint
-const commitTypes = [
-  'feat:',
-  'fix:',
-  'test:',
-  'ci:',
-  'chore:',
-  'docs:',
-  'refactor:',
-  'style:',
-  'perf:',
-  'revert:',
+const verbs = readFileSync(resolve(__dirname, '../data/verbs.txt'), 'utf8').split('\n');
+const approvedVerbs = [
+  'Add',
+  'Remove',
+  'Fix',
+  'Test',
+  'Document',
+  'Refactor',
+  'Style',
+  'Revert',
+  'Update',
+  'Configure',
+  'Deprecate',
+  'Correct',
+  'Improve',
+  'Initialise',
+  'Merge',
+  'Release',
 ];
 
 const linkForCommit = commit => `[${commit.sha.slice(0, 6)}](${commit.url})`;
-
 const commitMsg = prop('message');
-
-const startsWithI = needle => haystack => haystack.toLowerCase().startsWith(needle);
-const unknownPrefix = compose(none(Boolean), ap(map(startsWithI, commitTypes)), of);
-const commitsWithUnkownType =
-  filter(compose(unknownPrefix, commitMsg));
-
-const excludeMergeCommits =
-  reject(compose(startsWith('Merge'), commitMsg));
-
-const hasLinesOver = n => compose(
-  any(line => line.length > n),
-  split('\n'),
-);
-const commitsWithLinesOver = n =>
-  filter(compose(hasLinesOver(n), commitMsg));
-
-const reportCommits = (reporter, messageFn) => compose(
-  reporter,
-  mdList,
-  map(messageFn),
-);
-
-const maxCommitLineLength = 100;
-/**
- * Fails when any commit except Merge commits have a line in their body that exceeds 100 characters
- */
-exports.lineLength = () => {
-  const offendingCommits = compose(
-    commitsWithLinesOver(maxCommitLineLength),
-    excludeMergeCommits
-  )(danger.git.commits);
-
-  if (!isEmpty(offendingCommits)) {
-    reportCommits(fail, commit =>
-      `Commit ${linkForCommit(commit)} has lines with over ${maxCommitLineLength} chars.`
-    )(offendingCommits);
-  } else {
-    ok('lineLength');
+const excludeMergeCommits = reject(compose(startsWith('Merge'), commitMsg));
+
+exports.tests = {
+
+  commitApprovedVerb: {
+    citical: false,
+    test: pipe(
+      path(['git', 'commits']),
+      reject(pipe(commitMsg, split(' '), head, flip(contains)(approvedVerbs))),
+      map(linkForCommit),
+      map(commit => (
+        `Message for commit ${commit} should start with one of: ${approvedVerbs.join(', ')}.`
+      )),
+    ),
+  },
+
+  commitVerb: {
+    citical: true,
+    test: pipe(
+      path(['git', 'commits']),
+      reject(pipe(commitMsg, split(' '), head, flip(contains)(verbs))),
+      map(linkForCommit),
+      map(commit => (
+        `Message for commit ${commit} must start with an imperative verb.`
+      )),
+    ),
+  },
+
+  commitMessageLength: {
+    critical: true,
+    test: pipe(
+      path(['git', 'commits']),
+      excludeMergeCommits,
+      filter(pipe(commitMsg, split('\n'), any(x => x.length > 70))),
```

Comment:
Note that we were going to have **two** rules. One being a soft limit on 70 characters, and one being a hard limit on the column where GitHub breaks the line. Turns out GitHub breaks the line at 70 characters, so I did not include a soft limit rule.